app/screens/edit_profile/components/edit_profile_picture.tsx

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {Platform, StyleSheet, View} from 'react-native';
+import DocumentPicker from 'react-native-document-picker';
+
+import {Client} from '@client/rest';
+import ImagePicker from '@components/image_picker';
+import ProfileImage from '@components/profile_picture';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import useDidUpdate from '@hooks/did_update';
+import NetworkManager from '@init/network_manager';
+import UserModel from '@typings/database/models/servers/user';
+import {changeOpacity} from '@utils/theme';
+
+type ChangeProfilePictureProps = {
+    user: UserModel;
+    onUpdatedProfilePicture: (info: { localPath?: string; isRemoved?: boolean }) => void;
+};
+
+const SIZE = 128;
+
+const styles = StyleSheet.create({
+    container: {
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    camera: {
+        position: 'absolute',
+        overflow: 'hidden',
+        height: '100%',
+        width: '100%',
+    },
+});
+
+const EditProfilePicture = ({user, onUpdatedProfilePicture}: ChangeProfilePictureProps) => {
+    const [pictureUrl, setPictureUrl] = useState<string|undefined>();
+    const theme = useTheme();
+    const serverUrl = useServerUrl();
+
+    let client: Client | undefined;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch {
+        // does nothing
+    }
+
+    useEffect(() => {
+        // sets initial picture url
+        setPictureUrl(client?.getProfilePictureUrl(user.id, user.lastPictureUpdate));
+    }, []);
+
+    useDidUpdate(() => {
+        const url = user.id && client ? client.getProfilePictureUrl(user.id, user.lastPictureUpdate) : undefined;
+        if (url !== pictureUrl) {
+            setPictureUrl(url);
+        }
+    }, [user.id, user.lastPictureUpdate]);
+
+    const handleUploadProfileImage = useCallback((images: FileInfo[]) => {
+        const newImage = images?.[0]?.localPath;
+        if (newImage) {
+            setPictureUrl(newImage);
+            onUpdatedProfilePicture({isRemoved: false, localPath: newImage});
+        }
+    }, []);
```

Comment:
I haven't included it since `onUpdatedProfilePicture` is memoized, so we are sure to get the same reference each time; plus it also has no dependency.

Reply:
onUpdateProfilePicture is a prop, so potentially, it may be different.

If we are 100% sure that any caller to this component will only provide one single `onUpdatedProfilePicture`, we can leave it like this, but we may find in the future obscure complications if we start to change the `onUpdatedProfilePicture` prop.

Reply:
Adding it :)